### PR TITLE
Nelmio Alice v3 upgrade

### LINKED ISSUE #268: nelmio/alice v3
I apologize for the spammy issue, but as LiipFunctionalTestBundle is a one of the consumer of [alice](https://github.com/nelmio/alice), I though worth informing you that the next version [v3 is on it's way](https://github.com/nelmio/alice/issues/265). So if you noticed something you would like to change at alice level, now is a good time :)

### LINKED ISSUE #297: WebTest case is not working with nelmio alice 3.0
A class "Fixtures" is need by the method loadFixtures, Isuppose it has become FixtureSet in nelmio\alice 3.0.

I get this error:
"BadMethodCallException: nelmio/alice should be installed to use this method."

Any workaround?

Many thanks

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Closes #268
Closes #297 
Closes #321

- [x] Make the tests working
- [x] Update the readme according to the package requirements

@lsmith77 This change bump PHP requirement to **7.1** with nelmio usage.

Please tell me what you want for: https://github.com/liip/LiipFunctionalTestBundle/issues/268#issuecomment-340755081

I'll suggest to make it on a new major after waiting changes to clean other dependencies and bump our version support.

If you just think this change acceptable for a major, then merge it when ready and release it. :wink:

### CODE REVIEW

File: Test/WebTestCase.php [hunk 1]
```diff
@@ -552,16 +552,14 @@ private function locateResources($paths)
...
-            // @codeCoverageIgnoreStart
-            throw new \BadMethodCallException('nelmio/alice should be installed to use this method.');
-            // @codeCoverageIgnoreEnd
-        }
-
         /** @var ContainerInterface $container */
         $container = $this->getContainer();
 
+        $persisterLoaderServiceName = 'fidry_alice_data_fixtures.doctrine.persister_loader';
+        if (!$container->has($persisterLoaderServiceName)) {
```
Comment (Soullivaneuh) [inline Test/WebTestCase.php]: @theofidry This is always returning false even with the enabled bundle: https://github.com/liip/LiipFunctionalTestBundle/pull/337/files#diff-5658229c21833f6b720a1c07e3705aa0R26

Any idea?

Comment (Soullivaneuh) [reply Test/WebTestCase.php]: I know why: Wrong kernel. :troll:

**Comment by Soullivaneuh** [comment]
@lsmith77 As you can see the custom faker provider is still tested without hautelook and works like a charm. So we don't need it anymore.

Also, StyleCI is crying about unrelated file: https://styleci.io/analyses/qv5Dmd

Could you please make a fix on master so I can make a rebase? Thanks.

File: Test/WebTestCase.php [hunk 2]
```diff
@@ -574,18 +572,7 @@ public function loadFixtureFiles(array $paths = array(), $append = false, $omNam
 
         $files = $this->locateResources($paths);
 
-        // Check if the Hautelook AliceBundle is registered and if yes, use it instead of Nelmio Alice
-        $hautelookLoaderServiceName = 'hautelook_alice.fixtures.loader';
```
Comment (theofidry) [inline Test/WebTestCase.php]: I can't recall if that is purged, if it is then the above service should be changed for `fidry_alice_data_fixtures.loader.doctrine`

File: Tests/AppConfig/config.yml [hunk 3]
```diff
@@ -29,4 +29,4 @@ liip_functional_test:
 services:
     faker.provider.foo:
         class: Liip\FunctionalTestBundle\Tests\AppConfig\DataFixtures\Faker\Provider\FooProvider
-        tags: [ { name: hautelook_alice.faker.provider } ]
```
Comment (theofidry) [inline Tests/AppConfig/config.yml]: the doc needs to be updated accordingly

File: composer.json [hunk 4]
```diff
@@ -52,6 +51,8 @@
             "Liip\\FunctionalTestBundle\\": ""
         }
     },
+    "minimum-stability": "beta",
```
Comment (theofidry) [inline composer.json]: rc? (not sure that works)

File: composer.json [hunk 5]
```diff
@@ -30,11 +30,10 @@
         "doctrine/doctrine-fixtures-bundle": "~2.3",
         "phpunit/phpunit": "^4.8.35|^5.7|^6.1",
         "twig/twig": "~1.12|~2.0",
-        "nelmio/alice": "~1.7|~2.0",
+        "theofidry/alice-data-fixtures": "^1.0",
```
Comment (theofidry) [inline composer.json]: you also need to dump php to `^7.1` (requirement of `theofidry/alice-data-fixtures`)

File: Tests/AppConfig/config.yml (hunk 3 above)
Comment (Soullivaneuh) [reply Tests/AppConfig/config.yml]: See my todos.

File: composer.json (hunk 5 above)
Comment (Soullivaneuh) [reply composer.json]: I don't know, this is why I'm talking about a new major on the thread.

Indeed, PHP 7.1 is required for this dependency, but alice-data-fixtures is not on this bundle.

File: composer.json (hunk 4 above)
Comment (Soullivaneuh) [reply composer.json]: `RC`. I used `beta` before because I was required hautelook.

File: Test/WebTestCase.php (hunk 2 above)
Comment (Soullivaneuh) [reply Test/WebTestCase.php]: @theofidry You mean this? https://github.com/liip/LiipFunctionalTestBundle/pull/337/files#diff-f815866dba659cf4e7aebb6fc769c44cR558

Right?

Comment (theofidry) [reply Test/WebTestCase.php]: yep

**Comment by Soullivaneuh** [comment]
@lsmith77 I think this should be done on a new major with other updates, because it's finally BC break at end usage.

Indeed, the dependency is optional. So, someone making a `composer update` will get the next minor release of this bundle and will see all the tests suites break because of this "dependency" change.

Plus, if the concerned user is not under PHP 7.1, he would not be able to fix it and will have to change the liip dependency to `1.8.*` which is not semver compliant.

What do you think?

**Comment by Soullivaneuh** [comment]
@lsmith77 The PR is finished for the first part. I'm waiting Travis and your thinking about the possible next major release.

File: Test/WebTestCase.php (hunk 2 above)
Comment (Soullivaneuh) [reply Test/WebTestCase.php]: Well, there is nothing more to do in this case. :-)

File: Tests/AppConfig/config.yml (hunk 3 above)
Comment (Soullivaneuh) [reply Tests/AppConfig/config.yml]: @theofidry I removed the doc part concerning the faker provider. This part was only telling about hautelook installation which is not needed anymore.

**Comment by Soullivaneuh** [comment]
@theofidry Alice data-fixtures looks to not be compatible with SF 2.7: https://travis-ci.org/liip/LiipFunctionalTestBundle/jobs/295358260#L590

Could you confirm?

BTW, it looks like the bridge integration is not tested according to the dependencies: https://packagist.org/packages/nelmio/alice#v3.1.1

**Comment by theofidry** [comment]
>Alice data-fixtures looks to not be compatible with SF 2.7: https://travis-ci.org/liip/LiipFunctionalTestBundle/jobs/295358260#L590

It's not incompatible per se but that compatibility is not guaranteed neither tested for both AliceDataFixtures and NelmioAlice.

>BTW, it looks like the bridge integration is not tested according to the dependencies

There is a dep for local tests but for Travis the deps are overriden, so you can see it in the `.travis.yml`

**Comment by Soullivaneuh** [comment]
> It's not incompatible per se but that compatibility

So we also have to drop SF 2.7 dependency.

**Comment by theofidry** [comment]
Or add it back in the downstream libraries :)

But I'm more in favour of moving forward. We are bumping a bit everything and the old compatible stuff is not going away.

File: composer.json [hunk 6]
```diff
@@ -52,6 +51,8 @@
             "Liip\\FunctionalTestBundle\\": ""
         }
     },
+    "minimum-stability": "RC",
```
Comment (lsmith77) [inline composer.json]: why did you add this?

**Comment by lsmith77** [comment]
I also think this Bundle hasn't seen major updates in a long time, so I am not so worried about shedding some old stuff. That being said, we also talked about some other major refactorings to decouple the huge base class a bit.

File: composer.json (hunk 6 above)
Comment (theofidry) [reply composer.json]: theofidry/alice-data-fixtures is still in RC, so if not provided the user would need to install it himself and the require would not suffice

**Comment by Soullivaneuh** [comment]
@lsmith77 In this case I'll prepare some separated PR about version support dropping.

I assume you will create a 1.x branch and let master for 2.x?

**Comment by lsmith77** [comment]
yeah .. a 1.x branch makes sense .. you have the permissions to do this yourself right?

**Comment by Soullivaneuh** [comment]
@lsmith77 I don't think so, I'm not a maintainer of this bundle. :sweat_smile:

**Comment by Soullivaneuh** [comment]
@lsmith77 Please see: https://github.com/liip/LiipFunctionalTestBundle/issues/338

**Comment by alexislefebvre** [comment]
@Soullivaneuh I fixed the StyleCI warnings on the `master` branch. Rebasing your branch on the `master` branch should remove these warnings.

File: composer.json (hunk 5 above)
Comment (Jean85) [reply composer.json]: If we retarget this PR to the 2.x, requiring 7.1 is already included.

**Comment by Soullivaneuh** [comment]
@alexislefebvre I did a rebase, sorry for the delay.

**Comment by Jean85** [comment]
@Soullivaneuh you're still targeting master, that what you want to do? I already stated that I wouldn't do it...

**Comment by Soullivaneuh** [comment]
@Jean85 oh right! It's a mistake. I'm used to have master branch targeting unstable release. I'll rebase it.

File: composer.json (hunk 5 above)
Comment (Soullivaneuh) [reply composer.json]: Rebased.

**Comment by Jean85** [comment]
Ok, great! So we shall close this?

**Comment by Soullivaneuh** [comment]
We can.
